### ISSUE TITLE
SIGSEGV with RouteTableAssociation on pulumi-aws 4.11.0 and up

### ISSUE THREAD

**Issue by wimax-grapl** [pulumi/pulumi-aws#1575]
## Steps to reproduce
Sorry - I don't have a super tight repro for this.

Here's the requirements I'm working with, among others:
```
pulumi==3.7.1
pulumi-aws==4.11.0 # <----- 
pulumi-docker==3.0.0
pulumi-kafka==3.0.1
pulumi-random==4.2.0
```
Works fine with 4.10.0, breaks with 4.11 thru 4.13. # EDIT: MAY ALSO BE BROKEN WITH 4.10.0

The code that causes this error:
```
        for subnet in self.public_subnets:
            name = f"route-for-{subnet._name}"
            route_table = aws.ec2.RouteTable(
                name,
                vpc_id=self.vpc.id,
                routes=[
                    aws.ec2.RouteTableRouteArgs(
                        cidr_block="0.0.0.0/0",
                        gateway_id=self.internet_gateway.id,
                    )
                ],
                tags={"Name": name},
                opts=pulumi.ResourceOptions(parent=subnet),
            )

            aws.ec2.RouteTableAssociation( # <-------- This RouteTableAssociation
                f"assoc-{subnet._name}",
                subnet_id=subnet.id,
                route_table_id=route_table.id,
                opts=pulumi.ResourceOptions(parent=subnet),
            )
```

Expected: 
a `pulumi up` works without a SIGSEGV

Actual: 
```
 +  aws:ec2:RouteTable route-for-grapl-network-us-east-1b-public-subnet creating 
 +  aws:ec2:RouteTable route-for-grapl-network-us-east-1a-public-subnet created 
 +  aws:ec2:RouteTableAssociation assoc-grapl-network-us-east-1a-public-subnet creating 
 +  pulumi:pulumi:Stack grapl-local-grapl creating panic: runtime error: invalid memory address or nil pointer dereference
 +  pulumi:pulumi:Stack grapl-local-grapl creating [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x49c0bc3]
 +  pulumi:pulumi:Stack grapl-local-grapl creating goroutine 5398 [running]:
 +  pulumi:pulumi:Stack grapl-local-grapl creating github.com/terraform-providers/terraform-provider-aws/aws/internal/service/ec2/finder.RouteTableAssociationByID(0xc002585eb0, 0xc0016db848, 0x11, 0x2, 0x2, 0xc00280c5a0)
 +  pulumi:pulumi:Stack grapl-local-grapl creating      /home/runner/go/pkg/mod/github.com/pulumi/terraform-provider-aws@v1.38.1-0.20210706104737-8b9c0e66f8cb/aws/internal/service/ec2/finder/finder.go:295 +0x223
 +  pulumi:pulumi:Stack grapl-local-grapl creating github.com/terraform-providers/terraform-provider-aws/aws/internal/service/ec2/waiter.RouteTableAssociationState.func1(0xc000503ee0, 0xc000503ea0, 0x0, 0x0, 0x2, 0xc000503f01)
 +  pulumi:pulumi:Stack grapl-local-grapl creating      /home/runner/go/pkg/mod/github.com/pulumi/terraform-provider-aws@v1.38.1-0.20210706104737-8b9c0e66f8cb/aws/internal/service/ec2/waiter/status.go:291 +0x45
 +  pulumi:pulumi:Stack grapl-local-grapl creating github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.(*StateChangeConf).WaitForStateContext.func1(0xc0005b8840, 0xc000310150, 0xc0018ae420, 0xc0018a0c00, 0xc001691ef0, 0xc001691ee8)
 +  pulumi:pulumi:Stack grapl-local-grapl creating      /home/runner/go/pkg/mod/github.com/pulumi/terraform-plugin-sdk/v2@v2.0.0-20210402103405-f5979773e8ba/helper/resource/state.go:110 +0x2e9
 +  pulumi:pulumi:Stack grapl-local-grapl creating created by github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.(*StateChangeConf).WaitForStateContext
 +  pulumi:pulumi:Stack grapl-local-grapl creating      /home/runner/go/pkg/mod/github.com/pulumi/terraform-plugin-sdk/v2@v2.0.0-20210402103405-f5979773e8ba/helper/resource/state.go:83 +0x1c6
 +  aws:lambda:Function engagement-creator-Handler-lambda creating error: transport is closing
 +  aws:ec2:RouteTable route-for-grapl-network-us-east-1b-private-subnet creating error: transport is closing
 +  aws:ec2:RouteTableAssociation assoc-grapl-network-us-east-1a-public-subnet creating error: transport is closing
```

**Comment by wimax-grapl** [comment]
(NOTE: This may actually be breaking at 4.10)

**Comment by wimax-grapl** [comment]
I can't find the exact version of the file Pulumi is using, but here's where it is in latest terraform-provider-aws:

https://github.com/hashicorp/terraform-provider-aws/blob/main/aws/internal/service/ec2/finder/finder.go#L295

**Comment by leezen** [comment]
I suspect this may be related to https://github.com/hashicorp/terraform-provider-aws/issues/19985

**Comment by wimax-grapl** [comment]
Thanks for the quick followup 🙏  yeah deff looks like something upstream

**Comment by stack72** [comment]
This was fixed and released in v4.24.0 of the pulumi-aws provider 

Apologies for not posting that before
